tests/IBTK/samraidatacache_01.cpp

```diff
@@ -0,0 +1,244 @@
+            level->allocatePatchData(oec_int_idx, 0.0);
+            level->allocatePatchData(ofc_int_idx, 0.0);
+            level->allocatePatchData(onc_int_idx, 0.0);
+            level->allocatePatchData(osc_int_idx, 0.0);
+            level->allocatePatchData(sc_int_idx, 0.0);
+        }
+
+        {
+            const auto cc_double_cloned_idx = cached_data.getCachedPatchDataIndex(cc_double_idx);
+            const auto ec_double_cloned_idx = cached_data.getCachedPatchDataIndex(ec_double_idx);
+            const auto fc_double_cloned_idx = cached_data.getCachedPatchDataIndex(fc_double_idx);
+            const auto nc_double_cloned_idx = cached_data.getCachedPatchDataIndex(nc_double_idx);
+            const auto oec_double_cloned_idx = cached_data.getCachedPatchDataIndex(oec_double_idx);
+            const auto ofc_double_cloned_idx = cached_data.getCachedPatchDataIndex(ofc_double_idx);
+            const auto onc_double_cloned_idx = cached_data.getCachedPatchDataIndex(onc_double_idx);
+            const auto osc_double_cloned_idx = cached_data.getCachedPatchDataIndex(osc_double_idx);
+            const auto sc_double_cloned_idx = cached_data.getCachedPatchDataIndex(sc_double_idx);
+
+            const auto cc_float_cloned_idx = cached_data.getCachedPatchDataIndex(cc_float_idx);
+            const auto ec_float_cloned_idx = cached_data.getCachedPatchDataIndex(ec_float_idx);
+            const auto fc_float_cloned_idx = cached_data.getCachedPatchDataIndex(fc_float_idx);
+            const auto nc_float_cloned_idx = cached_data.getCachedPatchDataIndex(nc_float_idx);
+            const auto oec_float_cloned_idx = cached_data.getCachedPatchDataIndex(oec_float_idx);
+            const auto ofc_float_cloned_idx = cached_data.getCachedPatchDataIndex(ofc_float_idx);
+            const auto onc_float_cloned_idx = cached_data.getCachedPatchDataIndex(onc_float_idx);
+            const auto osc_float_cloned_idx = cached_data.getCachedPatchDataIndex(osc_float_idx);
+            const auto sc_float_cloned_idx = cached_data.getCachedPatchDataIndex(sc_float_idx);
+
+            const auto cc_int_cloned_idx = cached_data.getCachedPatchDataIndex(cc_int_idx);
+            const auto ec_int_cloned_idx = cached_data.getCachedPatchDataIndex(ec_int_idx);
+            const auto fc_int_cloned_idx = cached_data.getCachedPatchDataIndex(fc_int_idx);
+            const auto nc_int_cloned_idx = cached_data.getCachedPatchDataIndex(nc_int_idx);
+            const auto oec_int_cloned_idx = cached_data.getCachedPatchDataIndex(oec_int_idx);
+            const auto ofc_int_cloned_idx = cached_data.getCachedPatchDataIndex(ofc_int_idx);
+            const auto onc_int_cloned_idx = cached_data.getCachedPatchDataIndex(onc_int_idx);
+            const auto osc_int_cloned_idx = cached_data.getCachedPatchDataIndex(osc_int_idx);
+            const auto sc_int_cloned_idx = cached_data.getCachedPatchDataIndex(sc_int_idx);
+        }
```

Comment:
Could we somehow log the number of variables with allocated data before and after these handles go out of scope?

Reply:
Check it out now.

Reply:
Note that the registered indices do not go away until the caching object is deleted.